src/Microsoft.Azure.SignalR/EndpointProvider/ServiceEndpointManager.cs

```diff
@@ -30,5 +44,74 @@ public override IServiceEndpointProvider GetEndpointProvider(ServiceEndpoint end
 
             return new ServiceEndpointProvider(endpoint, _options);
         }
+
+        private void OnChange(ServiceOptions options)
+        {
+            Log.DetectConfigurationChanges(_logger);
+
+            // Reset local cache and validate result
+            SetValuableEndpoints(GetEndpoints(options));
+            if (Endpoints.Length == 0)
+            {
+                throw new ArgumentException(ServiceEndpointProvider.ConnectionStringNotFound);
```

Comment:
>ArgumentException [](start = 26, length = 17)

what's the behavior when OnChange throw?

Reply:
Good question. Skip changes and log error probably friendly? 

---
In reply to: [377512292](https://github.com/Azure/azure-signalr/pull/801#discussion_r377512292) [](ancestors = 377512292)